### ISSUE TITLE
Crash on Startup `cursors/coords.json could not be found`

### ISSUE THREAD

**Issue by cauterize** [anegostudios/VintageStory-Issues#3466]
### Game Version

v1.19.0

### Platform

Linux

### Modded

Vanilla

### SP/MP

Singleplayer

### How to reproduce

Download the Linux Tar and run install.sh

### Logs

```Running on 64 bit Linux (Nobara Linux 39 (KDE Plasma)) [Kernel 6.7.0.203] with 15888 MB RAM
Game Version: v1.19.0 (Stable)
18/01/2024 17.22.02: Critical error occurred
Loaded Mods: 
System.Exception: Asset game:textures/gui/cursors/coords.json could not be found
   at Vintagestory.Common.AssetManager.Get(AssetLocation Location) in VintagestoryLib\Common\FileIO\AssetManager.cs:line 274
   at Vintagestory.Common.AssetManager.Get(String Path) in VintagestoryLib\Common\FileIO\AssetManager.cs:line 266
   at Vintagestory.Client.ScreenManager.LoadCursor(String code) in VintagestoryLib\Client\ScreenManager.cs:line 327
   at Vintagestory.Client.ScreenManager.loadCursors() in VintagestoryLib\Client\ScreenManager.cs:line 315
   at Vintagestory.Client.ScreenManager.DoGameInitStage2() in VintagestoryLib\Client\ScreenManager.cs:line 214
   at Vintagestory.Client.GuiScreenLoadingGame.RenderToDefaultFramebuffer(Single dt) in VintagestoryLib\Client\MainMenu\Screens\GuiScreenLoadingGame.cs:line 51
   at Vintagestory.Client.ScreenManager.Render(Single dt) in VintagestoryLib\Client\ScreenManager.cs:line 732
   at Vintagestory.Client.ScreenManager.OnNewFrame(Single dt) in VintagestoryLib\Client\ScreenManager.cs:line 635
   at Vintagestory.Client.NoObf.ClientPlatformWindows.window_RenderFrame(FrameEventArgs e) in VintagestoryLib\Client\ClientPlatform\GameWindow.cs:line 111
   at OpenTK.Windowing.Desktop.GameWindow.OnRenderFrame(FrameEventArgs args)
   at OpenTK.Windowing.Desktop.GameWindow.Run()
   at Vintagestory.Client.ClientProgram.Start(ClientProgramArgs args, String[] rawArgs) in VintagestoryLib\Client\ClientProgram.cs:line 318
   at Vintagestory.Client.ClientProgram.<>c__DisplayClass9_0.<.ctor>b__1() in VintagestoryLib\Client\ClientProgram.cs:line 128
   at Vintagestory.ClientNative.CrashReporter.Start(ThreadStart start) in VintagestoryLib\Client\ClientPlatform\ClientNative\CrashReporter.cs:line 93
```

**Comment by cauterize** [comment]
> Does the file `assets/game/textures/gui/cursors/coords.json` exist in your game installation? It is present in mine ...

It did not. In fact the whole assets folder missed a lot of sub directories. I extracted them manually. Now it starts.

So I guess it's maybe an oversight in the `install.sh`. I will close this Issue.

**Comment by radfast** [comment]
No no, if it's a problem in the `install.sh` which we distribute with the Linux version download, then we need to know about it and fix it

**Comment by pingouyn** [comment]
when I tried the install.sh script on kubuntu 23.10 it always tried to install to /root  (I don't mean "/" it says "/root/...") instead of installing it somewhere in /home.  Not sure if it's related or not.

**Comment by cauterize** [comment]
I closed the Issue because the wiki stated that `instal.sh` is unstable anyway.

```
Apparently this install script doesn't work for everyone, so you may need to take these steps manually, or create the target install folder before running the script (may be a permissions issue?) For more details please refer to the Forum thread linked below.
```
[Source](https://wiki.vintagestory.at/index.php/Installing_the_game_on_Linux#:~:text=Apparently%20this%20install%20script%20doesn%27t%20work%20for%20everyone%2C%20so%20you%20may%20need%20to%20take%20these%20steps%20manually%2C%20or%20create%20the%20target%20install%20folder%20before%20running%20the%20script%20(may%20be%20a%20permissions%20issue%3F)%20For%20more%20details%20please%20refer%20to%20the%20Forum%20thread%20linked%20below.)

The install of 1.9.1 worked fine by the way. Really not sure that's happening...

**Comment by pingouyn** [comment]
Owwww you smart cookie! Could not remember, so I tried again installing version 1.19.1 using sudo and not using sudo.
"sudo ./install.sh" tries to install the game but fails (even though it says at the end of the installation verbose that it did): "Install game into /root/ApplicationData/vintagestory (data into /root/.config/VintagestoryData)".

"./install.sh" correctly installs the game in "/home/MyUserName/ApplicationData/vintagestory".  It also does not override the vm.max_map_count now, which is great, and correctly adds mesa_glthread to run.sh if selected during install.
Note that during the install I got the following error message (the line starting with "ls") though all files seem to be installed as they should in ApplicationData and in .config too:
Install game into /home/MyUserName/ApplicationData/vintagestory (data into /home/MyUserName/.config/VintagestoryData)
Create NewDir /home/MyUserName/ApplicationData
Create NewDir ./backup
ls: cannot access '/home/MyUserName/ApplicationData/vintagestory.*': No such file or directory
Move AppDir
(...)

So basically we need to check if sudo is used and let the user know to not use sudo, or need to change something in the install xscript between lines 8 and 12 I presume?

Question, why is the directory /home/User/ApplicationData created and VS is not installed in ~/User/.local/share/vintagestory or ~/.vintagestory, or else?  I am on Plasma from kde and didn't have this folder before it was needed for VS.  Genuine question as I am kinda new to linux, and surprised the ApplicationData directory is at the same level as Documents Images Music etc in /home.  Kinda look messy to my young penguin eyes.  IS it something to do with other distribution not having the same storage directories in /home or something?

**Comment by cauterize** [comment]
> @cauterize so with 1.19.1 (I assume not 1.9.1 ) you ran install.sh again and it worked without issues?

correct. (Non sudo)

**Comment by Th3Dilli** [comment]
@pingouyn 
we now check if it was run with side and abort

for the path, I have now overhauled the entire install.sh
and default to $HOME/.local/share/vintagestory
It has been ApplicationData presumable since when that was written this was then similar to how it was on windows but that does not make much sense on linux

also added some more option as well as an prompt to choose the install dir

since that has been brought up a already a few times

also added the install of the custom mime type handlers so when that is release linux also now has 1 click moddb install

so this should be solved now
ty everyone
